client/my-sites/stats/stats-notices/all-notice-definitions.ts

```diff
@@ -43,21 +43,30 @@ const ALL_STATS_NOTICES: StatsNoticeType[] = [
 			isSiteJetpackNotAtomic,
 			isCommercial,
 			hasPWYWPlanOnly,
+			shouldShowPaywallNotice,
 		}: StatsNoticeProps ) => {
+			// Show the notice only if the site is commercial.
+			if ( ! isCommercial ) {
+				return false;
+			}
+
+			// Show the upgrade notice with the coming paywall communication.
+			if ( shouldShowPaywallNotice ) {
```

Comment:
For sites with valid subscription, we should hide the notice right?

Reply:
Ah, you are right!

Reply:
Fixed in [3a26823](https://github.com/Automattic/wp-calypso/pull/92724/commits/3a2682359c0ba84682f26a99fbe3764026f5ea06).